src/dbus/dbus_thread_object.cpp

```diff
@@ -0,0 +1,449 @@
+
+    RegisterSetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_MESH_LOCAL_PREFIX_PROPERTY,
+                               std::bind(&DBusThreadObject::SetMeshLocalPrefixHandler, this, _1));
+    RegisterSetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_LEGACY_ULA_PREFIX_PROPERTY,
+                               std::bind(&DBusThreadObject::SetLegacyULAPrefixHandler, this, _1));
+    RegisterSetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_LINK_MODE_PROPERTY,
+                               std::bind(&DBusThreadObject::SetLinkModeHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_LINK_MODE_PROPERTY,
+                               std::bind(&DBusThreadObject::GetLinkModeHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_DEVICE_ROLE_PROPERTY,
+                               std::bind(&DBusThreadObject::GetDeviceRoleHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_NETWORK_NAME_PROPERTY,
+                               std::bind(&DBusThreadObject::GetNetworkNameHandler, this, _1));
+
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_PANID_PROPERTY,
+                               std::bind(&DBusThreadObject::GetPanIdHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_EXTPANID_PROPERTY,
+                               std::bind(&DBusThreadObject::GetExtPanIdHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_CHANNEL_PROPERTY,
+                               std::bind(&DBusThreadObject::GetChannelHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_MASTER_KEY_PROPERTY,
+                               std::bind(&DBusThreadObject::GetMasterKeyHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_CCA_FAILURE_RATE_PROPERTY,
+                               std::bind(&DBusThreadObject::GetCcaFailureRateHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_LINK_COUNTERS_PROPERTY,
+                               std::bind(&DBusThreadObject::GetLinkCountersHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_IP6_COUNTERS_PROPERTY,
+                               std::bind(&DBusThreadObject::GetIp6CountersHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_SUPPORTED_CHANNEL_MASK_PROPERTY,
+                               std::bind(&DBusThreadObject::GetSupportedChannelMaskHandler, this, _1));
+
+    return err;
+}
+
+void DBusThreadObject::DeviceRoleHandler(otDeviceRole aDeviceRole)
+{
+    SignalPropertyChanged(OTBR_DBUS_PREFIX, OTBR_DBUS_DEVICE_ROLE_PROPERTY, GetDeviceRoleName(aDeviceRole));
+}
+
+void DBusThreadObject::ScanHandler(DBusRequest &aRequest)
+{
+    auto &threadHelper = mNcp->GetThreadHelper();
+    if (threadHelper->Scan([aRequest](const std::vector<otActiveScanResult> &aResult) mutable {
```

Comment:
So why don't we design `ThreadHelper::Scan` as returning `void` and provide the guarantee that the parameter `aHandler` will always be called.